web/viewer.js

```diff
@@ -1300,6 +1300,35 @@ var PDFViewerApplication = {
 
     var body = document.querySelector('body');
     body.setAttribute('data-mozPrintCallback', true);
+
+    if (!this.hasEqualPageSizes) {
+      console.warn('Not all pages have the same size. The printed result ' +
+          'may be incorrect!');
+    }
+
+    // Insert a @page + size rule to make sure that the page size is correctly
+    // set. Note that we assume that all pages have the same size, because
```

Comment:
You could create a separate feature request for that, and link to some PDFs for testing. pages cannot be targeted separately, but your scenario could be addressed by rotating the pages before printing.

If I'm not mistaken, pages with alternating landscape/portrait modes are not printed correctly, right? If so, then the acceptance of this patch won't be a regression.